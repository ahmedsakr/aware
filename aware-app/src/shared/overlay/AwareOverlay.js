import React, {Component} from 'react'
import 'bootstrap'
import './AwareOverlay.css'

class AwareOverlay extends Component {

    render() {
        const {name, title, content, footer} = this.props;
    
        return(
            <div class="overlay-container">
                <div class="modal fade" id={name} tabindex="-1" role="dialog" aria-labelledby={name + "Label"} aria-hidden="true">
                    <div class="modal-dialog" role="document">
                        <div class="modal-content">
                            <div class="modal-header">
                                <div class="modal-title">{title}</div>
                                <button type="button" class="close" id="custom-overlay-close" data-dismiss="modal" aria-label="Close">
                                    <span aria-hidden="true">&times;</span>
                                </button>
                            </div>
                            <div class="modal-body">
                                {content}
                            </div>
                            <div class="modal-footer">
                                <div class="overlay-footer">
                                    {footer}
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default AwareOverlay;