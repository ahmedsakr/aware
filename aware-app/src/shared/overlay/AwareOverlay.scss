.overlay-container {
    display: flex;
}

.toggle-container {
    display: flex;
    flex: 1;
}

.overlay-dismiss {
    flex: 1;
    margin-right: 5px;
}

.overlay-footer {
    display: flex;
    flex: 1;
    justify-content: flex-end;
}

#custom-overlay-close {
    flex: 0;
}