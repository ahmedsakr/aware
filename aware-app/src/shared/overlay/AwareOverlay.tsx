import React from 'react'
import 'bootstrap'
import './AwareOverlay.scss'
import Modal from 'react-bootstrap/Modal'

type AwareOverlayProps = {
    name: string,
    title: string,
    content: JSX.Element,
    footer: JSX.Element
};

type AwareOverlayState = {
    show: boolean
};

export default class AwareOverlay extends React.Component<AwareOverlayProps, AwareOverlayState> {
    
    constructor(props: AwareOverlayProps) {
        super(props);

        this.state = {
            show: false
        };
    }

    show() : void {
        this.setState({show: true});
    }

    hide(): void {
        this.setState({show: false});
    }

    render(): JSX.Element {
        const { name, title, content, footer } = this.props;
        
        return (
            <div className="overlay-container">
                <Modal show={this.state.show}>
                    <Modal.Header closeButton>
                        <Modal.Title>{title}</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                        {content}
                    </Modal.Body>
                    <Modal.Footer>
                        <div className="overlay-footer">
                            {footer}
                        </div>                 
                    </Modal.Footer>
                </Modal>
            </div>
        )
    }
}
