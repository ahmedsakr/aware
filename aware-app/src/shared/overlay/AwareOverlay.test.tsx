import React from 'react';
import { shallow } from 'enzyme';

import AwareOverlay from './AwareOverlay';

describe('AwareOverlay', () => {
    it('should render correctly', () => {

        let content = () => {
            return (
                <div>
                    testing body content!
                </div>
            );
        }

        let footer = () => {
            return (
                <div>
                    testing footer content!
                </div>
            );
        }

        const component = shallow(
            <AwareOverlay
                name="Test overlay"
                title="Testing the overlay"
                content={content()}
                footer={footer()} />
        );

        expect(component).toMatchSnapshot();
    });
});