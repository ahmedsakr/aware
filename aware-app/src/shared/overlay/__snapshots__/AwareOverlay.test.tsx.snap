version https://git-lfs.github.com/spec/v1
oid sha256:605b54cf5773fe218f841c87c278bd65b1fedbcff48234da04e70db11ff0e1bc
size 624
